### PR TITLE
Reduce the default resource limits to minimum.

### PR DESCRIPTION
**Pull request by dmkozh**
# Description

Reduce the default resource limits to minimum.

Also refactor the test framework to have only a single way of modifying the config in tests (besides upgrades).

Even if we decide to go with higher limits initially, this change makes it as easy to update the limits without breaking tests.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/overlay/test/OverlayTests.cpp [hunk 1]
```diff
@@ -308,13 +315,14 @@ TEST_CASE("flow control byte capacity", "[overlay][flowcontrol]")
...
                 configEntry.contractBandwidth().ledgerMaxTxsSizeBytes =
-                    InitialSorobanNetworkConfig::
-                        LEDGER_MAX_TRANSACTION_SIZES_BYTES;
+                    maxTxSize * 10;
                 res = txtest::makeConfigUpgradeSet(ltx, configUpgradeSet);
                 ltx.commit();
             }
             txtest::executeUpgrade(*app, txtest::makeConfigUpgrade(*res));
         };
+        upgradeApp(app1, txSize);
```
Comment (marta-lokhova) [inline src/overlay/test/OverlayTests.cpp:324]: why is this needed if the config is overridden on line 163?

Comment (dmkozh) [reply src/overlay/test/OverlayTests.cpp:323]: Probably due to ledger-wide limit... Also this section concerns upgrades, so it would make sense to do the upgrade for the setup.

File: src/overlay/test/OverlayTests.cpp [hunk 2]
```diff
@@ -160,6 +157,16 @@ TEST_CASE("flow control byte capacity", "[overlay][flowcontrol]")
         auto app2 = createTestApplication(clock, cfg2, true, false);
         app1->getHerder().setMaxClassicTxSize(txSize);
         app2->getHerder().setMaxClassicTxSize(txSize);
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+        overrideSorobanNetworkConfigForTest(*app1);
+        overrideSorobanNetworkConfigForTest(*app2);
+        modifySorobanNetworkConfig(*app1, [txSize](SorobanNetworkConfig& cfg) {
```
[the remark this answers is not in the document]
Comment (dmkozh) [reply src/overlay/test/OverlayTests.cpp]: Done.

File: src/overlay/test/OverlayTests.cpp (hunk 1 above)
Comment (marta-lokhova) [reply src/overlay/test/OverlayTests.cpp:323]: this section does upgrades in a controlled way, where order and delay between upgrades matters (and upgrades trigger overlay events as well). Why not just set ledger-wide limit in the same lambda you're using above?

Comment (dmkozh) [reply src/overlay/test/OverlayTests.cpp:323]: The config overrides just override the config in-memory. This is a test util sufficient for most of the tests that don't care about the ledger state. As upgrades in fact do care about ledger state, I feel that it's more appropriate to prepare the proper ledger state first via the proper (upgrade) mechanism.

**Review by marta-lokhova — APPROVED** [review]

**Comment by sisuresh** [comment]
r+ ef847f0849575d6127efa855c34fc8e6c9cc383f
